discovery/dnsdisc/client.nim

```diff
@@ -77,13 +86,41 @@ proc resolveSubtreeEntry*(resolver: Resolver, loc: LinkEntry, subdomain: string)
     
   return ok(res[])
 
-proc resolveAllEntries(rootEntry: RootEntry): ResolveResult[seq[SubtreeEntry]] =
+proc resolveAllEntries*(resolver: Resolver, loc: LinkEntry, rootEntry: RootEntry): Future[seq[SubtreeEntry]] {.async.} =
   ## Resolves all subtree entries at given root
   ## Follows EIP-1459 client protocol
-  ## 
-  ## @TODO implement
   
-  ok(newSeq[SubtreeEntry]())
+  var subtreeEntries: seq[SubtreeEntry]
+  
+  # Initialise a hash set with the root hashes of ENR and link subtrees
+  var hashes = toHashSet([rootEntry.eroot, rootEntry.lroot])
+
+  while hashes.len > 0:
+    # Recursively resolve leaf entries and add to return list.
+    # @TODO: Should we add a depth limit here?
```

Comment:
I'd probably add a limit for safety, 100 or 1000 or so. Also, can recursion happen here? (beginning to sound like a linked list cycle interview problem :D)

Reply:
Good idea. Can fine tune later, but don't want to end-up trying to untangle Merkle ivy instead of a tree.

Reply:
Done in 27cfb5d